kornia/geometry/transform/crop/crop2d_contrib.py

```diff
@@ -0,0 +1,182 @@
+from typing import Optional
+
+import kornia
+import torch
+from .crop2d import crop_by_boxes as sampling_by_boxes, validate_bboxes
+
+
+def generate_dst_bbox_from_bbox(boxes: torch.Tensor):
+    """Generate 2D destination bounding boxes from bounding boxes. Preserving the exact width and height.
+
+    Args:
+        boxes (torch.Tensor): a tensor containing the coordinates of the bounding boxes to be extracted.
+            The tensor must have the shape of Bx4x2, where each box is defined in the following (clockwise)
+            order: top-left, top-right, bottom-right and bottom-left. The coordinates must be in the x, y order.
+
+    Returns:
+        torch.Tensor: the output mask tensor.
+
+    Examples:
+        >>> boxes = torch.tensor([[
+        ...        [1., 1.],
+        ...        [4., 1.],
+        ...        [4., 2.],
+        ...        [1., 2.],
+        ...   ], [
+        ...        [1., 3.],
+        ...        [8., 3.],
+        ...        [8., 7.],
+        ...        [1., 7.],
+        ...   ]])  # 1x4x2
+        >>> generate_dst_bbox_from_bbox(boxes)
+        tensor([[[0., 0.],
+                 [3., 0.],
+                 [3., 1.],
+                 [0., 1.]],
+        <BLANKLINE>
+                [[0., 0.],
+                 [7., 0.],
+                 [7., 4.],
+                 [0., 4.]]])
+    """
+    dst_w = boxes[:, 1, 0] - boxes[:, 0, 0]
+    dst_h = boxes[:, 3, 1] - boxes[:, 1, 1]
+    zeros = torch.zeros_like(dst_w)
+
+    points_dst: torch.Tensor = torch.stack([
+        torch.stack([zeros, zeros], dim=-1),
+        torch.stack([dst_w, zeros], dim=-1),
+        torch.stack([dst_w, dst_h], dim=-1),
+        torch.stack([zeros, dst_h], dim=-1)
+    ], dim=1)
+    return points_dst
+
+
+def bbox_to_mask(boxes: torch.Tensor, mask: torch.Tensor) -> torch.Tensor:
```

Comment:
What the use case of this function ? from the description I don't understand the functionality and the purpose.